src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -338,7 +338,11 @@ public function find($id, $columns = ['*'])
      */
     public function findMany($ids, $columns = ['*'])
     {
-        if (empty($ids)) {
+        $empty = $ids instanceof Arrayable
+            ? empty($ids->toArray())
+            : empty($ids);
+
+        if ($empty) {
```

Comment:
It's better if you cast the ids and then do the empty check in the if statement.

```php
	        $ids = $ids instanceof Arrayable ? $ids->toArray()) : $ids;

	        if (empty($ids)) {
```

That way it's also only two lines you're adding.

Reply:
It's done. I didn't want before to change argument value passed to further calls.